app/scripts/modules/core/src/pipeline/config/stages/checkPreconditions/CheckPreconditionsExecutionDetails.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { get } from 'lodash';
+
+import { StageFailureMessage } from 'core/delivery/details';
+import { ExecutionDetailsSection, IExecutionDetailsSectionProps } from '../core';
+import { robotToHuman } from 'core/presentation/robotToHumanFilter/robotToHuman.filter';
+
+export function CheckPreconditionsExecutionDetails(props: IExecutionDetailsSectionProps) {
+  const context = get(props.stage, 'context.context', {} as any);
+  return (
+    <ExecutionDetailsSection name={props.name} current={props.current}>
+      <div className="row">
+        <div className="col-md-12">
+          <dl className="dl-horizontal">
+            {Object.keys(context).filter((key) => key !== 'expression' && context[key] !== null).map((key) => (
+              <div key={key}>
+                <dt>{robotToHuman(key)}</dt>
+                <dd>{JSON.stringify(context[key])}</dd>
+              </div>
+            ))}
+            <div>
+              <dt>Fail Pipeline</dt>
+              <dd>{props.stage.context.failPipeline ? 'true' : 'false'}</dd>
```

Comment:
`<dd>{!!props.stage.context.failPipeline}}</dd>` ?

Reply:
One would hope. Alas, JSX does not stringify the Boolean value. I could do `String(!!props.stage.context.failPipeline)` though...